### ISSUE TITLE
cmake tools 1.9.2 for linux  uses -H option to specify source dir

### ISSUE THREAD

**Issue by enderson-pan** [microsoft/vscode-cmake-tools#2291]
### Brief Issue Summary
CMake tools 1.9.2, VSCode 1.63.2, Ubuntu 20.04 LTS
When execute CMake:Configure from VSCode Command Palette，the output shows that cmake tools use cmake -H option to specify the source dir. In my opinion, mybe -S should be used?

```
[proc] Executing command: /usr/bin/cmake --no-warn-unused-cli -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_C_COMPILER:FILEPATH=/bin/x86_64-linux-gnu-gcc-9 -DCMAKE_CXX_COMPILER:FILEPATH=/bin/x86_64-linux-gnu-g++-9 -H/home/sake/workspace/study-cmake/cmake-workshop/content/code/day-1/01_libraries-cxx -B/home/sake/workspace/study-cmake/cmake-workshop/content/code/day-1/01_libraries-cxx/build -G "Unix Makefiles"
```

Even if I specify the version of the cmake in CMakeLists.txt, nothing changes.
```
cmake_minimum_required(VERSION 3.15...3.22)
```

### CMake Tools Diagnostics

```
<!-- Paste the `CMake: Log Diagnostics` output HERE -->
{
  "os": "linux",
  "vscodeVersion": "1.63.2",
  "cmtVersion": "1.9.2",
  "configurations": [
    {
      "folder": "/home/sake/workspace/study-cmake/cmake-workshop/content/code/day-1/01_libraries-cxx",
      "cmakeVersion": "3.22.1",
      "configured": true,
      "generator": "Unix Makefiles",
      "usesPresets": false,
      "compilers": {
        "C": "/bin/x86_64-linux-gnu-gcc-9",
        "CXX": "/bin/x86_64-linux-gnu-g++-9"
      }
    }
  ],
  "cpptoolsIntegration": {
    "isReady": true,
    "hasCodeModel": true,
    "activeBuildType": "Debug",
    "buildTypesSeen": [
      "Debug"
    ],
    "requests": [
      "file:///home/sake/workspace/study-cmake/cmake-workshop/content/code/day-1/01_libraries-cxx/hello-world.cpp"
    ],
    "responses": [
      {
        "uri": "file:///home/sake/workspace/study-cmake/cmake-workshop/content/code/day-1/01_libraries-cxx/hello-world.cpp",
        "configuration": {
          "defines": [],
          "includePath": [],
          "compilerPath": "/bin/x86_64-linux-gnu-g++-9",
          "compilerArgs": [
            "-g"
          ]
        }
      }
    ],
    "partialMatches": [],
    "targetCount": 2,
    "executablesCount": 1,
    "librariesCount": 1,
    "targets": [
      {
        "name": "hello-world",
        "type": "EXECUTABLE"
      },
      {
        "name": "message",
        "type": "STATIC_LIBRARY"
      }
    ]
  },
  "settings": [
    {
      "communicationMode": "automatic",
      "useCMakePresets": "auto",
      "configureOnOpen": true
    }
  ]
}
```

### Debug Log

```
<!-- Paste the debug log contents HERE -->
[main] Configuring folder: 01_libraries-cxx 
[main] Saving open files before configure/build
[driver] Run _refreshExpansions
[driver] Run _refreshExpansions cb
[driver] Start configure 
[driver] Runnnig pre-configure checks and steps
[driver] Using compilers in GCC 9.3.0 x86_64-linux-gnu for configure
[driver] Run _refreshExpansions
[driver] Run _refreshExpansions cb
[cmakefileapi-driver] Configuring using kit
[cmakefileapi-driver] Invoking CMake /usr/bin/cmake with arguments ["--no-warn-unused-cli","-DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE","-DCMAKE_BUILD_TYPE:STRING=Debug","-DCMAKE_C_COMPILER:FILEPATH=/bin/x86_64-linux-gnu-gcc-9","-DCMAKE_CXX_COMPILER:FILEPATH=/bin/x86_64-linux-gnu-g++-9","-H/home/sake/workspace/study-cmake/cmake-workshop/content/code/day-1/01_libraries-cxx","-B/home/sake/workspace/study-cmake/cmake-workshop/content/code/day-1/01_libraries-cxx/build","-G","Unix Makefiles"]
[proc] Executing command: /usr/bin/cmake --no-warn-unused-cli -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_C_COMPILER:FILEPATH=/bin/x86_64-linux-gnu-gcc-9 -DCMAKE_CXX_COMPILER:FILEPATH=/bin/x86_64-linux-gnu-g++-9 -H/home/sake/workspace/study-cmake/cmake-workshop/content/code/day-1/01_libraries-cxx -B/home/sake/workspace/study-cmake/cmake-workshop/content/code/day-1/01_libraries-cxx/build -G "Unix Makefiles"
[cmake] Not searching for unused variables given on the command line.
[cmake] -- Configuring done
[cmake] -- Generating done
[cmake] -- Build files have been written to: /home/sake/workspace/study-cmake/cmake-workshop/content/code/day-1/01_libraries-cxx/build
[cmakefileapi-parser] Read reply folder: /home/sake/workspace/study-cmake/cmake-workshop/content/code/day-1/01_libraries-cxx/build/.cmake/api/v1/reply
[cmakefileapi-parser] Found index files: ["cache-v2-2b6dae4bcdfdebea7676.json","codemodel-v2-13f095d4ae0c7800426c.json","directory-.-Debug-f5ebdc15457944623624.json","index-2021-12-17T10-08-14-0095.json","target-hello-world-Debug-6420117a78027635b0e4.json","target-message-Debug-6c719e2f4e8d7bab0da7.json","toolchains-v1-76f98403275c4e665006.json"]
[cache] Reading CMake cache file /home/sake/workspace/study-cmake/cmake-workshop/content/code/day-1/01_libraries-cxx/build/CMakeCache.txt
[cache] Parsing CMake cache string
[extension] [4381] cmake.configure finished (returned 0)
```

**Comment by friendlyanon** [comment]
This is #2053 and will be fixed once #2292 is merged.

**Comment by bobbrow** [comment]
#2292 is merged.
